core/src/main/java/io/github/fourlastor/game/demo/turns/AttackRanged.java

```diff
@@ -0,0 +1,125 @@
+package io.github.fourlastor.game.demo.turns;
+
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.scenes.scene2d.actions.Actions;
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+import com.badlogic.gdx.scenes.scene2d.ui.Image;
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedFactory;
+import dagger.assisted.AssistedInject;
+import io.github.fourlastor.game.demo.state.GameState;
+import io.github.fourlastor.game.demo.state.unit.Unit;
+
+public class AttackRanged extends TurnState {
+
+    private final StateRouter router;
+
+    private final Unit source;
+    private final Unit target;
+    Vector2 originalPosition = new Vector2();
+    float scale = 1f; // Amount to scale the animation by.
+    int damage = 1;
+    TextureAtlas textureAtlas;
+
+    @AssistedInject
+    public AttackRanged(@Assisted Attack attack, @Assisted TextureAtlas textureAtlas, StateRouter router) {
+        this.router = router;
+        this.source = attack.source;
+        this.target = attack.target;
+        this.textureAtlas = textureAtlas;
+        this.originalPosition.set(source.getActorPosition());
+    }
+
+    /**
+     * TODO: This functionality can be used generically. Unsure where to move it to.
+     *
+     * Calculate the angle in degrees between two 2D vectors.
+     *
+     * @param source The source vector.
+     * @param target The target vector.
+     * @return The angle in degrees between the vectors.
+     */
+    public static float calculateAngle(Vector2 source, Vector2 target) {
+        float deltaX = target.x - source.x;
+        float deltaY = target.y - source.y;
+
+        float angleRad = MathUtils.atan2(deltaY, deltaX);
+        float angleDeg = MathUtils.radiansToDegrees * angleRad;
+
+        source.angleDeg(target);
+
+        return angleDeg;
+    }
```

Comment:
you can simplify this, see https://github.com/fourlastor-jams/into-the-wip/pull/12/commits/bb4ae1fcc3c7ddf89d30285ec3fe09191fb7e285